TimoCloud-Universal/src/main/java/cloud/timo/TimoCloud/base/managers/BaseInstanceManager.java

```diff
@@ -305,19 +309,63 @@ private void startServer(BaseServerObject server) {
                 throw new ServerStartException("Could not start process");
             }
 
-            TimoCloudBase.getInstance().getSocketMessageManager().sendMessage(Message.create()
-                    .setType(MessageType.BASE_SERVER_STARTED)
-                    .setTarget(server.getId())
-                    .set("port", serverPort)
-                    .set("publicKey", RSAKeyUtil.publicKeyToBase64(publicKey))
-            );
-
+            //delay because we can only get the pid after 20 Millis.
+            new java.util.Timer().schedule(
+                    new java.util.TimerTask() {
+                        @Override
+                        public void run() {
+                            try {
+                                int pid = findScreenPid(server.getId());
+                                TimoCloudBase.getInstance().getSocketMessageManager().sendMessage(Message.create()
+                                        .setType(MessageType.BASE_SERVER_STARTED)
+                                        .setTarget(server.getId())
+                                        .set("port", serverPort)
+                                        .set("publicKey", RSAKeyUtil.publicKeyToBase64(publicKey))
+                                        .set("pid", pid)
+                                );
+                            } catch (Exception e) {
+                                TimoCloudBase.getInstance().severe("Error while starting server " + server.getName() + ": " + e.getMessage());
+                                TimoCloudBase.getInstance().getSocketMessageManager().sendMessage(Message.create().setType(MessageType.BASE_SERVER_NOT_STARTED).setTarget(server.getId()));
+                            }
+                        }
+                    }, 20);
```

Comment:
I wouldn't hardcode a limit because this could depend on the system load. Rather, I would do it in a loop (with a maximum amount of retries), always waiting 20ms between the tries.